src/coreclr/jit/fgdiagnostic.cpp

```diff
@@ -3148,6 +3148,73 @@ void Compiler::fgDebugCheckFlags(GenTree* tree)
             expectedFlags |= (GTF_GLOB_REF | GTF_ASG);
             break;
 
+#if defined(FEATURE_HW_INTRINSICS)
+        case GT_HWINTRINSIC:
+        {
+            GenTreeHWIntrinsic* hwintrinsic = tree->AsHWIntrinsic();
+            NamedIntrinsic      intrinsicId = hwintrinsic->GetHWIntrinsicId();
+
+            if (hwintrinsic->OperIsMemoryStore())
+            {
+                assert(tree->OperRequiresAsgFlag());
+                assert(tree->OperMayThrow(this));
```

Comment:
```suggestion
```
Redundant with the logic above that adds the relevant flags into `expectedFlags`.

Reply:
This is asserting that those cases were hit here. Otherwise you could theoretically end up with a `OperIsMemoryStore` that is not marked as `GTF_ASG` or `GTF_EXCEPT`